Objective-C/Tests/RemoteDatabaseTest.m

```diff
@@ -0,0 +1,235 @@
+//
+//  RemoteDatabaseTest.m
+//  CouchbaseLite
+//
+//  Copyright (c) 2022 Couchbase, Inc All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+#import "RemoteDatabaseTest.h"
+#import "CBLRemoteDatabase.h"
+#import "CBLErrors.h"
+
+@implementation RemoteDatabaseTest {
+    CBLRemoteDatabase* _client;
+}
+
+#pragma mark - helper methods
+
+- (void) startConnectedClient: (nullable NSURL*)url {
+    _client = [[CBLRemoteDatabase alloc] initWithURL: url
+                                        authenticator: nil];
+}
+
+- (void) validateDocument: (CBLDocument*)expDoc
+                errorCode: (NSInteger)errorCode  {
+    XCTestExpectation* e = [self expectationWithDescription: @"validation exp"];
+    __block NSInteger code = errorCode;
+    [_client documentWithID: expDoc.id completion:^(CBLDocument* doc, NSError* error) {
+        if (code != 0) {
+            AssertEqual(error.code, code);  // error code as expected
+            AssertNil(doc);                 // empty doc in case of error.
+        } else {
+            AssertNil(error);               // empty error
+
+                                            // same doc-information returned
+            AssertEqualObjects(doc.id, expDoc.id);
+            AssertEqualObjects([doc toDictionary], [expDoc toDictionary]);
+        }
+        [e fulfill];
+    }];
+    
+    [self waitForExpectations: @[e] timeout: timeout];
+}
+
+#pragma mark - lifecycle
+
+- (void) setUp {
+    [super setUp];
+    timeout = 10.0;
+}
+
+- (void) tearDown {
+    if (_client)
+        [_client stop];
+    _client = nil;
+    
+    [self stopListen];
+    [super tearDown];
+}
+
+#pragma mark - Tests
+
+- (void) testConnectedClient {
+    CBLDatabase.log.console.level = kCBLLogLevelDebug;
```

Comment:
This shouldn't be here otherwise the rest of the tests will run with debug log level.